tools/ingest_historical_csv.py

```diff
@@ -0,0 +1,82 @@
+'''Ingests historical CSV data into DB'''
+
+import argparse
+import asyncio
+import logging
+import textwrap
+
+from aiohttp import ClientSession
+import pandas as pd
+import daiquiri
+
+daiquiri.setup(level=logging.INFO)
+log = daiquiri.getLogger(__name__) # pylint: disable=invalid-name
+
+async def _ingest_probable_cve(df, session: ClientSession, url): # pylint: disable=invalid-name
+    obj = df.to_dict(orient='records')
+    async with session.post(url, json=obj) as response:
+        log.info('Got response {} for {}'.format(response.status, url))
+
+async def _add_feedback(df, session: ClientSession, url): # pylint: disable=invalid-name
+    if len(df) < 1:
+        return
+    df['author'] = 'anonymous'
+    df['feedback_type'] = 'POSITIVE'
+    # typo carried away
+    for feedback in ('Feeedback', 'Feedback'):
+        if feedback in df:
+            df[feedback].fillna('', inplace=True)
+            df.rename(columns=dict([(feedback, 'comments')]), inplace=True)
+            break
+    else:
+        df['comments'] = ''
+    df = df[['author', 'feedback_type', 'comments', 'url']]
+    # df['comments'] = ''
+    obj = df.to_dict(orient='records')
+    async with session.post(url, json=obj) as response:
+        log.info('Got response {} for {}'.format(response.status, url))
+
+def _get_executor(args):
+    if args.feedback:
+        return _add_feedback, args.feedback
+    return _ingest_probable_cve, args.insert
+
+async def _main(args):
+    log.info('invoking ingestion for {} CSV files'.format(len(args.csv)))
+    func, url = _get_executor(args)
+    for csv in args.csv:
+        log.info('Convert records in {} to JSON'.format(csv))
+        df = pd.read_csv(csv, index_col=None, header=0) # pylint: disable=invalid-name
+        log.info('Ingest {} records to DB'.format(len(df)))
+        async with ClientSession() as session:
+            await func(df=df, session=session, url=url)
```

Comment:
Keep in mind about duplicate records if you run multiple concurrent request. In place of this you can run sequentially all request or further split panda dataframe to multiple dataframes based on uniqueness constrain and then can run few in parallel that we know it wont cause any issue.

Reply:
It is better to impose it in server side based on [sticky sessions](https://docs.openshift.com/container-platform/3.6/architecture/networking/routes.html#routes-sticky-sessions), then it will become a single solution for both live ingestion and historical ingestion.

Reply:
sure we can do that.